core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt

```diff
@@ -153,48 +153,48 @@ object GeneUtils {
     }
 
     fun applyJsonEscapes(string: String, format: OutputFormat = OutputFormat.JAVA_JUNIT_4):String{
-        val ret = when{
-            format.isKotlin() -> string.replace("\$", "\\\$")
-            else -> string
-        }
-        return ret.replace("\\", """\\""")
+        val ret = string.replace("\\", """\\""")
                 .replace("\"", "\\\"")
                 .replace("\n", "\\n")
                 .replace("\r", "\\r")
                 .replace("\b", "\\b")
                 .replace("\t", "\\t")
 
+        when{
+            format.isKotlin() -> return ret.replace("\$", "\\\$")
+            else -> return ret
+        }
+
     }
 
     fun applyUriEscapes(string: String, format: OutputFormat = OutputFormat.JAVA_JUNIT_4):String{
```

Comment:
I m bit unsure of this method. there is much more that need encoding, and some of it is already handled when evaluating the fitness function.
look at  `RestPath.encode`, `RestPath.resolveOnlyPath` and `AbstractRestFitness.handleRestCall`. In particular, in this latterthe following will likely need to be removed / updated:

```
 .let {
                    /*
                        TODO this will be need to be done properly, and check if
                        it is or not a valid char.
                        Furthermore, likely needed to be done in resolveLocation,
                        or at least check how RestAssured would behave
                     */
                    it.replace("\"", "")
                }
```